apps/web/src/components/BridgeForm.tsx

```diff
@@ -130,28 +130,48 @@ export default function BridgeForm({
   const { getBalance } = useCheckBalance();
   const [isBalanceSufficient, setIsBalanceSufficient] = useState(true);
   const [tokenBalances, setTokenBalances] = useState({});
+  const [isVerifyingTransaction, setIsVerifyingTransaction] = useState(false);
 
-  const checkBalance = debounce(async () => {
+  async function getBalanceFn() {
     const key = `${selectedNetworkA.name}-${selectedTokensA.tokenB.symbol}`;
     const balance = await getBalance(selectedTokensA.tokenB.symbol);
-    setTokenBalances({
+    const updatedBalances = {
       ...tokenBalances,
       [key]: balance,
-    });
+    };
+
+    setTokenBalances(updatedBalances);
+    return updatedBalances;
+  }
+
+  const checkBalance = debounce(async () => {
+    getBalanceFn();
   }, 200);
 
-  useEffect(() => {
+  async function verifyTransfer(refetch?: boolean) {
     const key = `${selectedNetworkA.name}-${selectedTokensA.tokenB.symbol}`;
-    const balance = tokenBalances[key];
+    const refetchBalance = refetch ? await getBalanceFn() : tokenBalances;
+    const balance = refetch ? refetchBalance[key] : tokenBalances[key];
```

Comment:
```suggestion
   const balance = (refetch ? await getBalanceFn() : tokenBalances)[key];
```
hmm is this better? I noticed that the second ternary operator isn't necessary

Reply:
haha preference :v :v i guess